MapboxDirections/MBDirectionsOptions.swift

```diff
@@ -0,0 +1,457 @@
+        case "text":
+            format = .text
+        case "html":
+            format = .html
+        default:
+            return nil
+        }
+        self.init(rawValue: format.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .text:
+            return "text"
+        case .html:
+            return "html"
+        }
+    }
+}
+
+/**
+ Options for calculating results from the Mapbox Directions service.
+ 
+ You do not create instances of this class directly. Instead, create instances of `MatchOptions` or `RouteOptions`.
+ */
+@objc(MBDirectionsOptions)
+open class DirectionsOptions: NSObject, NSSecureCoding, NSCopying {
+    
+    /**
+     Initializes a route options object for routes between the given waypoints and an optional profile identifier.
+     
+     - parameter waypoints: An array of `Waypoint` objects representing locations that the route should visit in chronological order. The array should contain at least two waypoints (the source and destination) and at most 25 waypoints. (Some profiles, such as `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, [may have lower limits](https://www.mapbox.com/api-documentation/#directions).)
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. This parameter, if set, should be set to `MBDirectionsProfileIdentifierAutomobile`, `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, `MBDirectionsProfileIdentifierCycling`, or `MBDirectionsProfileIdentifierWalking`. `MBDirectionsProfileIdentifierAutomobile` is used by default.
+     */
+    @objc required public init(waypoints: [Waypoint], profileIdentifier: MBDirectionsProfileIdentifier? = nil) {
+        self.waypoints = waypoints
+        self.profileIdentifier = profileIdentifier ?? .automobile
+    }
+    
+    public class var supportsSecureCoding: Bool {
+        return true
+    }
+    
+    // MARK: NSCopying
+    @objc open func copy(with zone: NSZone? = nil) -> Any {
+        let copy = type(of: self).init(waypoints: waypoints, profileIdentifier: profileIdentifier)
+        copy.includesSteps = includesSteps
+        copy.shapeFormat = shapeFormat
+        copy.routeShapeResolution = routeShapeResolution
+        copy.locale = locale
+        copy.attributeOptions = attributeOptions
+        copy.includesSpokenInstructions = includesSpokenInstructions
+        copy.distanceMeasurementSystem = distanceMeasurementSystem
+        copy.includesVisualInstructions = includesVisualInstructions
+        return copy
+    }
+    
+    //MARK: - OBJ-C Equality
+    open override func isEqual(_ object: Any?) -> Bool {
+        guard let opts = object as? DirectionsOptions else { return false }
+        return isEqual(to: opts)
+    }
+    
+    @objc(isEqualToDirectionsOptions:)
+    open func isEqual(to directionsOptions: DirectionsOptions?) -> Bool {
+        guard let other = directionsOptions else { return false }
```

Comment:
This method should insist that the two objects are of exactly the same class. Otherwise, you can ask a MatchOptions if it’s equal to a RouteOptions, and it might say yes. That might be handy in some cases, but it’s better to do that sort of equivalence checking explicitly rather than relying on an equality method.

/ref https://github.com/mapbox/MapboxDirections.swift/pull/236#discussion_r172036916